app/threads.py

```diff
@@ -0,0 +1,11 @@
+from threading import Thread
+from boutiques import search
+import json
+
+
+class UpdatePipelineData(Thread):
+
+    def run(self):
+        with open(".cache/boutiques/pipelineData.json", "w") as f:
+            data = search("-m 1000", "-nt")
+            json.dump(data, f, indent=4)
```

Comment:
You should also pull all the descriptors and use the info inside to provide a more detailed view if the user clicks on one.